core/src/main/scala/kafka/coordinator/transaction/TransactionLog.scala

```diff
@@ -241,60 +255,11 @@ object TransactionLogManager {
   }
 }
 
-case class TxnKey(version: Short, key: String) extends BaseKey {
-
-  override def toString: String = key.toString
+trait BaseKey{
```

Comment:
nit: with only one message type in the log, this `BaseKey` no longer makes much sense. It's pretty small, so I'm OK leaving it, but we could just have `readMessageKey` return `TxnKey` directly and avoid the match on the result.